Firestore/Source/Core/FSTView.mm

```diff
@@ -485,26 +487,26 @@ - (void)applyTargetChange:(nullable FSTTargetChange *)targetChange {
 
 @end
 
-static inline int DocumentViewChangeTypePosition(FSTDocumentViewChangeType changeType) {
+static inline int DocumentViewChangeTypePosition(DocumentViewChangeType changeType) {
   switch (changeType) {
-    case FSTDocumentViewChangeTypeRemoved:
+    case DocumentViewChangeType::Removed:
       return 0;
-    case FSTDocumentViewChangeTypeAdded:
+    case DocumentViewChangeType::Added:
       return 1;
-    case FSTDocumentViewChangeTypeModified:
+    case DocumentViewChangeType::Modified:
       return 2;
-    case FSTDocumentViewChangeTypeMetadata:
+    case DocumentViewChangeType::Metadata:
       // A metadata change is converted to a modified change at the public API layer. Since we sort
       // by document key and then change type, metadata and modified changes must be sorted
       // equivalently.
       return 2;
     default:
-      HARD_FAIL("Unknown FSTDocumentViewChangeType %s", changeType);
+      HARD_FAIL("Unknown DocumentViewChangeType %s", changeType);
```

Comment:
Rather than using a default state, move the HARD_FAIL outside the switch. (This will allow the compiler to fail when we add a new state.) i.e.:

```
switch (changeType) {
  case Rm: return 0;
  case Add: return 1;
  case Mod: return 2;
  case Meta: return 2;
}
HARD_FAIL(...);
```

Hmmm... although, that assumes the obj-c compiler will notice. (Our c++ compilers do.) Even if it doesn't, this might still be worthwhile under the assumption that this will eventually be a cc file.

Reply:
sgtm